src/custom/hooks/useDebounceWithForceUpdate.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react'
+
+// modified from https://usehooks.com/useDebounce/
+export default function useDebounceWithForceUpdate<T>(
+  value: T,
+  delay: number,
+  forceUpdateRef?: string | number | boolean
+): T {
+  const [lastForceUpdateRef, setLastForceUpdateRef] = useState(forceUpdateRef)
+  const [debouncedValue, setDebouncedValue] = useState<T>(value)
+
+  // force update
+  useEffect(() => {
+    if (
+      (Boolean(value) && debouncedValue === undefined) ||
+      (Boolean(forceUpdateRef) && forceUpdateRef !== lastForceUpdateRef)
+    ) {
+      setDebouncedValue(value)
+      setLastForceUpdateRef(forceUpdateRef)
+    }
+  }, [debouncedValue, forceUpdateRef, lastForceUpdateRef, value])
+
+  // Update debounced value after delay
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value)
```

Comment:
Don't you need to set the last forced value? although I think we probably don't need two states...